it-ops-shared/ITOps.ViewModelComposition/Modules/RequestsModule.cs

```diff
@@ -0,0 +1,139 @@
+                handlers = new List<Handler>();
+                gets.Add(template, handlers);
+            }
+
+            handlers.Add(new Handler() { onExecute = get, onError = onError });
+        }
+
+        //bool IInterceptRoutes.Matches(RouteData routeData, string httpVerb, HttpRequest request)
+        //{
+        //    RouteData = routeData;
+        //    HttpRequest = request;
+
+        //    switch (httpVerb.ToLowerInvariant())
+        //    {
+        //        case "get":
+        //            {
+        //                if (!getsCache.TryGetValue(request.Path, out Handler[] handlers))
+        //                {
+        //                    handlers = gets.Where(kvp => routeMatcher.Match(kvp.Key, request.Path, routeData.DataTokens))
+        //                        .SelectMany(kvp => kvp.Value)
+        //                        .ToArray();
+        //                    getsCache.Add(request.Path, handlers);
+        //                }
+
+        //                return handlers.Any();
+        //            }
+
+        //        default:
+        //            return false;
+        //    }
+        //}
+
+        //Task IHandleRequests.Handle(string requestId, dynamic vm, RouteData routeData, HttpRequest request)
+        //{
+        //    switch (request.Method.ToLowerInvariant())
+        //    {
+        //        case "get":
+        //            {
+        //                var handlers = getsCache[request.Path];
+        //                var pendingTasks = new List<Task>();
+        //                foreach (var item in handlers)
+        //                {
+        //                    pendingTasks.Add(item.onExecute(requestId, vm));
+        //                }
+
+        //                return Task.WhenAll(pendingTasks);
+        //            }
+        //        default:
+        //            throw new NotSupportedException();
+        //    }
+        //}
+
+        //Task IHandleRequestsErrors.OnRequestError(string requestId, Exception ex, dynamic vm, RouteData routeData, HttpRequest request)
+        //{
+        //    RouteData = routeData;
+        //    HttpRequest = request;
+
+        //    switch (request.Method.ToLowerInvariant())
+        //    {
+        //        case "get":
+        //            {
+        //                var handlers = getsCache[request.Path];
+        //                var pendingTasks = new List<Task>();
+        //                foreach (var item in handlers)
+        //                {
+        //                    pendingTasks.Add(item.onError(requestId, ex, vm));
+        //                }
+
+        //                return Task.WhenAll(pendingTasks);
+        //            }
+        //        default:
+        //            throw new NotSupportedException();
+        //    }
+        //}
+    }
+
+    public class RouteMatcher
+    {
+        public bool Match(string routeTemplate, string requestPath, RouteValueDictionary routeValueDictionary)
```

Comment:
Can we use .net core routing implementation for this?

Reply:
yes, but not in this spike. The spike was intended only to investigate syntax and feasibility. To use the default ASP.Net Core routing engine more changes are required to the composition engine itself. It can be done, and it's my intention to go ahead in my spare time.

Reply:
Sure! excuse my enthusiasm :)